src/Admin/BlockAdmin.php

```diff
@@ -378,4 +432,42 @@ protected function configureFormFields(FormMapper $formMapper): void
             ;
         }
     }
+
+    private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
+    {
+        $service = $this->blockManager->get($block);
+        $blockType = $block->getType();
+
+        if (!$service instanceof BlockServiceInterface) {
+            throw new \RuntimeException(sprintf(
```

Comment:
IMO, `UnexpectedValueException` or `DomainException` seem more accurate for this case.

Reply:
This is just a copy & paste of https://github.com/sonata-project/SonataPageBundle/pull/1064/files?file-filters%5B%5D=.php